kratos/utilities/sub_model_part_entities_boolean_operation_utility.cpp

```diff
@@ -0,0 +1,126 @@
+    const std::vector<IndexType>& rIds,
+    ModelPart& rModelPart)
+{
+    rModelPart.AddConditions(rIds);
+}
+
+template<class TEntityType, class TContainerType>
+std::vector<IndexType> SubModelPartEntitiesBooleanOperationUtility<
+    TEntityType, TContainerType>::GetContainerIds(ModelPart& rModelPart)
+{
+    const TContainerType& r_container = GetContainer(rModelPart);
+    std::vector<IndexType> ids_vector(r_container.size());
+    IndexPartition<std::size_t>(r_container.size()).for_each([&](std::size_t i){
+        ids_vector[i] = (r_container.begin()+i)->Id();
+    });
+    return ids_vector;
+}
+
+template<class TEntityType, class TContainerType>
+void SubModelPartEntitiesBooleanOperationUtility<
+    TEntityType, TContainerType>::BooleanOperation(
+    ModelPart& rModelPartA,
+    ModelPart& rModelPartB,
+    ModelPart& rDestination,
+    BooleanOperators ThisOperator)
+{
+    KRATOS_ERROR_IF(!rDestination.IsSubModelPart()) << "The destination model part must be a sub model part." << std::endl;
+    const ModelPart& r_root_a = rModelPartA.GetRootModelPart();
+    const ModelPart& r_root_b = rModelPartB.GetRootModelPart();
+    const ModelPart& r_root_d = rDestination.GetRootModelPart();
+    KRATOS_ERROR_IF(&r_root_a != &r_root_b) << "The first and second model parts must belong to the same root model part." << std::endl;
+    KRATOS_ERROR_IF(&r_root_a != &r_root_d) << "The destination model part must belong to the same root model part than the first and the second." << std::endl;
+    std::vector<IndexType> ids_a = GetContainerIds(rModelPartA);
+    std::vector<IndexType> ids_b = GetContainerIds(rModelPartB);
+    std::vector<IndexType> ids_destination;
+    std::sort(ids_a.begin(), ids_a.end());
+    std::sort(ids_b.begin(), ids_b.end());
+
+    if (ThisOperator == BooleanOperators::Union) {
+        std::set_union(
+            ids_a.begin(), ids_a.end(),
+            ids_b.begin(), ids_b.end(),
+            std::back_inserter(ids_destination));
+    } else if (ThisOperator == BooleanOperators::Intersection) {
+        std::set_intersection(
+            ids_a.begin(), ids_a.end(),
+            ids_b.begin(), ids_b.end(),
+            std::back_inserter(ids_destination));
+    } else if (ThisOperator == BooleanOperators::Difference) {
+        std::set_difference(
+            ids_a.begin(), ids_a.end(),
+            ids_b.begin(), ids_b.end(),
+            std::back_inserter(ids_destination));
+    }
+
+    EntitiesEraseProcess<TEntityType>(rDestination, EntitiesEraseProcessFlags::ERASE_ALL_ENTITIES).Execute();
+    AddEntities(ids_destination, rDestination);
+}
+
+template class SubModelPartEntitiesBooleanOperationUtility<Node<3>,ModelPart::NodesContainerType>;
+template class SubModelPartEntitiesBooleanOperationUtility<Element,ModelPart::ElementsContainerType>;
+template class SubModelPartEntitiesBooleanOperationUtility<Condition,ModelPart::ConditionsContainerType>;
+}  // namespace Kratos.
```

Comment:
can you please complete it to also work with MasterSlaveConstraints?

```suggestion
template class SubModelPartEntitiesBooleanOperationUtility<MasterSlaveConstraint,ModelPart::MasterSlaveConstraintContainerType>;
}  // namespace Kratos.
```